File: Bank-backend/src/main/java/com/ipnet/Bank_backend/services/CompteService.java
package com.ipnet.Bank_backend.services;

import com.ipnet.Bank_backend.entities.Compte;

import java.util.List;

public interface CompteService {

    Compte create (Compte compte);
    Compte update (Compte compte, Long id);
    List<Compte>  listeCompte ();
    void  delete (Long id);
    Compte  findById (Long id);
}

File: vue-crud-app/src/services/CompteService.js
import http from "../http-common";

class CompteService {
  getAll() {
    return http.get("/compte/");
  }

  get(id) {
    return http.get(`/compte/${id}`);
  }

  create(data) {
    return http.post("/compte/", data);
  }

  update(id, data) {
    return http.put(`/compte/${id}`, data);
  }

  delete(id) {
    return http.delete(`/compte/${id}`);
  }

  deleteAll() {
    return http.delete(`/compte`);
  }

  findByTitle(title) {
    return http.get(`/compte?title=${title}`);
  }
}

export default new CompteService();
